src/Fixer/LanguageConstruct/FunctionToConstantFixer.php

```diff
@@ -0,0 +1,202 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\LanguageConstruct;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException;
+use PhpCsFixer\Fixer\ConfigurableFixerInterface;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\Token;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class FunctionToConstantFixer extends AbstractFixer implements ConfigurableFixerInterface
+{
+    /**
+     * @var array<string, string>
+     */
+    private static $defaultConfiguration = array(
+        'phpversion' => 'PHP_VERSION',
+        'php_sapi_name' => 'PHP_SAPI',
+        'pi' => 'M_PI',
+    );
+
+    /**
+     * @var array<string, string>
+     */
+    private $configuration;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        if (null === $configuration) {
+            $this->configuration = self::$defaultConfiguration;
+
+            return;
+        }
+
+        $this->configuration = array();
+        foreach ($configuration as $key => $value) {
+            if (!is_string($key) || !array_key_exists($key, self::$defaultConfiguration)) {
+                throw new InvalidFixerConfigurationException($this->getName(), sprintf('Configuration key must be any of "%s".', implode('", "', array_keys(self::$defaultConfiguration))));
+            }
+
+            if (!is_bool($value)) {
+                throw new InvalidFixerConfigurationException(
+                    $this->getName(),
+                    sprintf(
+                        'Configuration value must be an bool, got "%s" for "%s".',
+                        is_object($value) ? get_class($value) : (gettype($value).'#'.$value),
+                        $key
+                    )
+                );
+            }
+
+            if ($value) {
```

Comment:
>Here we had this: ['foo' => 'FOO', 'bar' => 'BAR']

This was never used as the configuration format for this fixer.

The configuration here was;
`['funcA', 'funcB', ...  ]`
In the configure method this was mapped to
`['funcA' => 'funcA replacement', 'funcB'  => 'funcA replacement', ...  ]`

Now this is replaced with, configuration:
`['funcA' => bool, 'funcB' => bool, ...  ]`
Mapping; to the same format

The proposed `array_merge` will not result in the same mapping.

>Thinking again, I think boolean options here are not the best solution because you may still mix "mapping" and "behavior" options

I see no point in the booleans either, passing the function names did the trick just as well, however it was requested to change it; "Or store the map internally and use boolean options:" (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2486#issuecomment-286410636). 
Restructuring the config format for new options (so not new replacements) can be done when needed.

>and you lose the ability to choose the constants.

Nothing is lost, as it was never added. It is not a generic find-replace-function-with-constant-fixer by design.